server_addon/houdini/client/ayon_houdini/plugins/publish/collect_cache_farm.py

```diff
@@ -27,47 +27,35 @@ def process(self, instance):
         # Why do we need this particular collector to collect the expected
         # output files from a ROP node. Don't we have a dedicated collector
         # for that yet?
+        # Answer: No, we don't have a generic expected file collector.
+        #         Because different product types needs different logic.
+        #         e.g. check CollectMantraROPRenderProducts
+        #               and CollectKarmaROPRenderProducts
         # Collect expected files
         ropnode = hou.node(instance.data["instance_node"])
         output_parm = lib.get_output_parameter(ropnode)
         expected_filepath = output_parm.eval()
         instance.data.setdefault("files", list())
         instance.data.setdefault("expectedFiles", list())
-        if instance.data.get("frames"):
-            files = self.get_files(instance, expected_filepath)
-            # list of files
-            instance.data["files"].extend(files)
-        else:
+
+        frames = instance.data.get("frames", "")
+        if isinstance(frames, str):
             # single file
-            instance.data["files"].append(output_parm.eval())
-        cache_files = {"_": instance.data["files"]}
-        # Convert instance family to pointcache if it is bgeo or abc
-        # because ???
-        self.log.debug(instance.data["families"])
+            instance.data["files"].append(expected_filepath)
+        else:
+            # list of files
+            staging_dir, _ = os.path.split(expected_filepath)
+            instance.data["files"].extend(
+                ["{}/{}".format(staging_dir, f) for f in frames]
+            )
+
+        cache_files = {"cache": instance.data["files"]}
```

Comment:
It feels a bit odd that we need both the `instance.data["files"]` and `instance.data["expectedFiles"]` - they seem to want to do the same thing but for whatever reason we require the duplicated data. Might be good for a follow-up issue to streamline that logic over time?

Reply:
I think your explanation here helps a lot - but this should definitely be a "cleanup" issue so that we clean this up over time because as long as there's no reason for both being separate we should towards (once again) streamlining the API to just need a list of files with whatever metadata it may need downstream. Whether that is local, farm, or split renders or whatever.

Reply:
I checked it again. sorry for using this comment to reply to multiple conversations on this PR.

`CollectDataforCache` plugin is simply doing
```
instance.data["expectedFiles"] = {"cache": instance.data["files"]}
```
which follows the same logic as render products i.e. `{"aov": list_of_files}` which in turn allows us to re-use the deadline submitter with small tweaks to make it work with cache.

However, `CollectFilesForCleaningUp` doesn't care whether or not it's render or a cache. doesn't care whether or not it's local or farm. 
It follows a general rule. always check `expectedFiles`  then `frames` then fail to default `filepath`.
`filepath` must exist because, we are using it to compute the staging directory. 
and therefore, such a warning exists

https://github.com/ynput/ayon-core/blob/c2d851a9b578f563b8c4b6eb91923abad5025ace/server_addon/houdini/client/ayon_houdini/plugins/publish/collect_files_for_cleaning_up.py#L49-L51

Reply:
since it works I'll leave it be and I guess we'll just track a "cleanup" issue for the future when ayon houdini repo is separated.